### PR TITLE
Added a filterQuality parameter to texture

### PR DESCRIPTION
**Pull request by zljj0818**
## Description
Framework code for https://github.com/flutter/engine/pull/18814

## Related Issues
https://github.com/flutter/flutter/issues/53080

## Tests

I added the following tests:
widgets/texture_test.dart

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by zljj0818** [comment]
@liyuqian  I start a new PR here, for i do something wrong in the original PR:(

**Review by liyuqian — COMMENTED** [review]
LGTM with a few nits on documentations. CC @goderbauer to see if all his comments in the old PR are addressed too.

File: packages/flutter/lib/src/widgets/texture.dart
Comment (zljj0818) [reply packages/flutter/lib/src/widgets/texture.dart]: Done

File: packages/flutter/lib/src/widgets/texture.dart
Comment (zljj0818) [reply packages/flutter/lib/src/widgets/texture.dart]: Done

**Comment by goderbauer** [comment]
Looks like a test was added. I'll defer to @liyuqian for a detailed review then.

**Review by liyuqian — APPROVED** [review]

**Comment by flutter-github-sync** [comment]
Started Google testing for this PR

**Comment by flutter-github-sync** [comment]
Google testing failed...

**Comment by liyuqian** [comment]
The "Google testing" is failing because this PR depend on an engine PR https://github.com/flutter/engine/pull/18814. That test won't pass until the engine PR rolled into Google.

**Comment by zljj0818** [comment]
> The "Google testing" is failing because this PR depend on an engine PR [flutter/engine#18814](https://github.com/flutter/engine/pull/18814). That test won't pass until the engine PR rolled into Google.

:(  The Google testing will be triggered automatically when the engine PR rolled into Google?

**Comment by liyuqian** [comment]
It's a new test (just added last week) that will be triggered whenever a PR is approved.

**Comment by zljj0818** [comment]
> It's a new test (just added last week) that will be triggered whenever a PR is approved.

OK, when this PR can be merged?

**Comment by zljj0818** [comment]
Can try the google test again？

**Comment by liyuqian** [comment]
Of course, @renyou retried it yesterday. I see it pass now. Adding "waiting for tree to go green" label so this will be auto-merged once the "flutter-build" is green.

**Pull request merged**
